### PR TITLE
docs: fix semantic version link

### PR DESCRIPTION
**Pull request by hongbo-miao**

### CODE REVIEW

**Comment by chriseth** [comment]
Thanks for the link change! If I remember correctly, there was more information on the old page. I'm wondering if that can still be found somewhere...

**Comment by chriseth** [comment]
Old page: https://web.archive.org/web/20170312140408/https://docs.npmjs.com/misc/semver

**Comment by chriseth** [comment]
Maybe we should link here? https://github.com/npm/node-semver/blob/master/README.md

**Comment by cameel** [comment]
I don't think we support the whole range of version specifiers provided by npm. For example `~1.2.3`, `1.2.*`, `1.2.3 - 2.3.4` or even crazy stuff like  `>=1.2.3 <1.(2+1).0`.

I think we would be better off just describing our (much smaller) subset of that syntax.

**Comment by cameel** [comment]
Anyway, that would be a separate issue.

Regarding just the link, looks like the old page is still available here: https://docs.npmjs.com/cli/v6/using-npm/semver

**Comment by axic** [comment]
> I think we would be better off just describing our (much smaller) subset of that syntax.

To be fair we support a fairly large ruleset we should not support :wink:

I do agree we'd be better off describing ours and linking to upstream as inspiration.

File: docs/layout-of-source-files.rst [hunk 1]
```diff
@@ -79,7 +79,7 @@ be sure that your code compiles the way you intended. The exact version of the
 compiler is not fixed, so that bugfix releases are still possible.
 
 It is possible to specify more complex rules for the compiler version,
-these follow the same syntax used by `npm <https://docs.npmjs.com/misc/semver>`_.
+these follow the same syntax used by `npm <https://docs.npmjs.com/about-semantic-versioning>`_.
```
Comment (cameel) [inline docs/layout-of-source-files.rst:82]:
```suggestion
these follow the same syntax used by `npm <https://docs.npmjs.com/cli/v6/using-npm/semver>`_.
```

Comment (cameel) [inline docs/layout-of-source-files.rst]:
```suggestion
these follow the same syntax used by `npm <https://docs.npmjs.com/cli/v6/using-npm/semver>`_.
```

**Review by cameel — CHANGES_REQUESTED** [review]

**Comment by hongbo-miao** [comment]
Thanks! Yeah, I think it makes sense, the latest version doc is not very detailed like the old one. Updated as suggestion.

**Comment by cameel** [comment]
@Hongbo-Miao Could you also rebase on the current `develop`? There was an issue with windows build and it's fixed in `develop` but CI checks here won't pass until you rebase.

**Review by cameel — APPROVED** [review]
Some macOS CI jobs are failing but I don't think this is related to the change here.

**Pull request merged**